docs/source/tutorials/staging_data.ipynb

```diff
@@ -1,20 +1,340 @@
 {
```

Comment:
<ul><li>"for pilot" -&gt; for a pilot</li><li>"within <code>PilotDescription</code> or as input parameters in <code>pilot.stage_in()</code> method" -&gt; within <code>PilotDescription</code> or as input parameters in the <code>pilot.stage_in()</code> method</li><li>"but staging of output data is aplicable by using <code>pilot.stage_out()</code> method only." -&gt; but staging of output data can only be done by using the <code>pilot.stage_out()</code> method.</li></ul><p> </p><p>Why and when should we use the pilot data staging capabilities and/or the task staging capabilities? For example, is there anything a user a user should be considering from a performance/efficiency point of view? How does this capability help a user when writing a scientific application? What are the scenarios in which this capability is useful? The same comment applies to all the staging capabilities described here. User-facing documentation is not just a list of what is available.</p><p> </p><p>We are missing a discussion about the performance issues related to staging data. That is especially important when discussing staging between two hots. We need to explain that file staging will 'eat' into the walltime of the pilot job and that users need to be mindful about then when designing their applications. For example, 10K tasks, each with 5MB input file to transfer = likely a problem. We also need to explain that staging out will be truncated/stopped when the pilot exits so staging out 10GB at the end of a 20h simulation is likely going to fail. And so on.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/radical-cybertools/radical.pilot/pull/2882/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>